pkg/bpf/map.go

```diff
@@ -114,6 +115,42 @@ type MapInfo struct {
 	OwnerProgType ProgType
 }
 
+// DesiredAction is the action to be performed on the BPF map
+type DesiredAction int
+
+const (
+	// OK indicates that to further action is required and the entry is in
+	// sync
+	OK DesiredAction = iota
+
+	// Insert indicates that the entry needs to be created or updated
+	Insert
+
+	// Delete indicates that the entry needs to be deleted
+	Delete
+)
+
+func getActionName(action DesiredAction) string {
+	switch action {
+	case OK:
+		return "sync"
+	case Insert:
+		return "to-be-inserted"
```

Comment:
These action names are odd to me. Why "To be inserted"? Isn't the key already inserted in the bpf map? Do we have a middle step that is executed before actually putting the key in the map?

Reply:
@aanm "OK" represents that the desired state has been met. The "action" here, that needs to be performed, is an Insert or a Delete.

See line 509 of this same file; the DesiredAction is Insert or Delete if the operation did not complete successfully. Maybe renaming this function to "getDesiredActionName" would make this clearer.